Add hot reload when running yarn dev

sheeley opened the pull request:
For blacksmithgu#411

This adds a test vault which includes the hot reload plugin, and changes
`yarn dev`'s build location to `test-vault/.obsidian/plugins/dataview`.

blacksmithgu reviewed (changes requested):
Looks good - I would exclude `.obsidian/plugins/dataview/*` since that is a build artifact which will change and shouldn't be checked in. I'd also delete the `./obsidian/workspace` file.

sheeley commented:
Cleaned up and added copying to the rollup config. I kept `.hotreload` and the directory, as the other plugin depends upon its existence and it seemed silly to copy an empty file.
